VisualPinball.Unity/VisualPinball.Unity.Editor/Utils/TreeViewWithTreeModel/TreeElement.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+
+namespace VisualPinball.Unity.Editor.Utils.TreeViewWithTreeModel
+{
+
+	/// <summary>
+	/// TreeElement is the base class for creating TreeModel which will be used by TreeViewWithTreeModel, you'll have to inherit from it.
+	/// It provides base properties to be inserted & identified in a TreeView (Id, Name, Depth)
+	/// It hosts also its hierarchical information.
+	/// Some properties are serializable to save tree structure as Assets.
+	/// </summary>
+	[Serializable]
+	public class TreeElement
+	{
+		[SerializeField]
+		public int Id { get; set; }
+		[SerializeField]
+		public virtual string Name { get; }
+
+		[SerializeField]
+		public int Depth { get; set; }

```

Comment:
I think the `[Serializable]` on top of the class makes all members serializable per default, so only the `[NonSerialized]` attributes are needed.